src/lib.rs

```diff
@@ -115,6 +118,51 @@ mod errors {
 }
 pub use errors::*;
 
+/// A builder for configuring a `Mapping`.
+///
+/// ```
+/// # fn foo() -> addr2line::Result<()> {
+/// use addr2line::Options;
+///
+/// let mapping = Options::default()
+///     .with_functions()
+///     .build("path/to/some/executable")?;
+/// # let _ = mapping;
+/// # }
+/// ```
+#[derive(Clone, Copy, Default)]
+pub struct Options {
+    with_functions: bool,
+    with_demangling: bool,
+}
+
+impl Options {
+    /// Make the `Mapping` also include debug information for functions,
+    /// enabling `Mapping::locate()` to also indicate what function a given
+    /// address appears in. This comes at some parsing and lookup cost.
+    pub fn with_functions(mut self) -> Self {
+        self.with_functions = true;
+        self
+    }
+
+    /// Make the `Mapping` attempt to demangle Rust and C++ symbols. This option
+    /// implies `with_functions`.
+    pub fn with_demangling(mut self) -> Self {
+        self.with_demangling = true;
+        self.with_functions()
+    }
+
+    /// Finish configuration and build the `Mapping`.
+    ///
+    /// The target file will be memmap'd, and then `gimli` is used to parse out
+    /// the necessary debug symbols, without copying data when possible.
+    pub fn build<P>(self, file_path: P) -> Result<Mapping>
```

Comment:
One option here would be to have `Options::new`, which requires `file_path` (and not implement `Default`), and then implement `Into<Mapping>` for `Options`.